Lib/fontTools/cffLib/CFFToCFF2.py

```diff
@@ -101,9 +102,66 @@ def _convertCFFToCFF2(cff, otFont):
             if hasattr(topDict, key):
                 delattr(topDict, key)
 
+    # Clean up T2CharStrings
+
+    charStrings = topDict.CharStrings
+    globalSubrs = cff.GlobalSubrs
+    localSubrs = [getattr(fd.Private, "Subrs", []) for fd in fdArray]
+
+    for glyphName in charStrings.keys():
+        cs, fdIndex = charStrings.getItemAndSelector(glyphName)
+        cs.decompile()
+
+    # Clean up subroutines first
+    for subrs in [globalSubrs] + localSubrs:
+        for subr in subrs:
+            program = subr.program
+            i = j = len(program)
+            try:
+                i = program.index("return")
+            except ValueError:
+                pass
+            try:
+                j = program.index("endchar")
+            except ValueError:
+                pass
+            program[min(i, j) :] = []
+
+    # Clean up glyph charstrings
+    for glyphName in charStrings.keys():
+        cs, fdIndex = charStrings.getItemAndSelector(glyphName)
+        program = cs.program
+        if fdIndex == None:
+            fdIndex = 0
+
+        # Intentionally use None for nominalWidthX, such that any
+        # CharString that has an explicit width encoded will throw.
+        extractor = T2WidthExtractor(localSubrs[fdIndex], globalSubrs, None, 0)
+        try:
+            extractor.execute(cs)
+        except TypeError:
```

Comment:
TypeError seems a bit too broad here, I wonder if the T2WidthExtractor could throw some more specific exception that you can catch here?

Reply:
Yeah I don't like it either. I'll make a custom object that throws a custom exception on addition. No need to change the `T2WidthExtractor`.

Reply:
The merger uses a custom object as defaultWidth instead and detects if that was returned as width. But this way using exceptions stops as soon as the question is answered so I went with this model this time.

Reply:
i'm simply concerned that a TypeError may trigger inadvertently for unrelated bugs, activating the subrs-embeddindg-width-popping logic below